synapse/logging/_remote.py

```diff
@@ -54,10 +54,10 @@ class LogProducer:
 
     # This is essentially ITCPTransport, but that is missing certain fields
     # (connected and registerProducer) which are part of the implementation.
-    transport = attr.ib(type=Connection)
-    _format = attr.ib(type=Callable[[logging.LogRecord], str])
-    _buffer = attr.ib(type=deque)
-    _paused = attr.ib(default=False, type=bool, init=False)
+    transport: Connection
+    _format: Callable[[logging.LogRecord], str]
+    _buffer: deque
```

Comment:
Can we make this hint a [`typing.Deque`](https://docs.python.org/3/library/typing.html?highlight=typing%20deque#typing.Deque) so we can specify the type of its elements?